extensions/avro/src/main/java/com/hazelcast/jet/avro/AvroSerializerHooks.java

```diff
@@ -156,6 +144,20 @@ public GenericContainer read(@Nonnull ObjectDataInput in) throws IOException {
         public boolean isOverwritable() {
             return false;
         }
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance instance) {
```

Comment:
The internals of this setter are not obvious. Usually, setters don't have side-effects for the argument. It's better to document the fact that this method is used as a listener registration.

Reply:
[`e1c3fd70..b132d684`][1]

[1]: https://github.com/hazelcast/hazelcast/compare/e1c3fd70d93fb0b4956297b6031479d37166855f..b132d684268595d3468902788ffb8a7e401c6df6